fundamentals-section/server/main.go

```diff
@@ -1,20 +1,180 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
+	"log"
 	"net/http"
+	"os"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-// welcomeHandler is a handler function.
-func welcomeHandler(w http.ResponseWriter, req *http.Request) {
-	fmt.Fprintf(w, "Hello, friends! Welcome to the Microsoft Tech Days with Liam & Adelina!\n")
+// Person represents a person document in the database
+type Person struct {
+	ID       primitive.ObjectID `bson:"_id,omitempty" json:"id"`
+	Name     string             `bson:"name" json:"name"`
+	Fact     string             `bson:"fact" json:"fact"`
+	Location string             `bson:"location" json:"location"`
 }
 
 func main() {
-	// Register the welcomeHandler function to serve the root endpoint.
-	http.HandleFunc("/", welcomeHandler)
+	// Set the MongoDB connection URI, including the username, password, and database name
+	uri := os.Getenv("MONGODB_CONNECTION_STRING")
+
+	// Set the client options
+	clientOptions := options.Client().ApplyURI(uri)
+
+	// Set the context with a 10-second timeout
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Connect to the MongoDB instance
+	client, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Check the connection
+	err = client.Ping(ctx, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("Connected to Cosmos DB MongoDB instance!")
+
+	http.HandleFunc("/persons", func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case "GET":
+			getAllPeople(w, r, client)
+		case "POST":
+			createPerson(w, r, client)
+		case "OPTIONS":
+			// Set necessary headers for handling CORS requests
+			w.Header().Set("Access-Control-Allow-Origin", "*")
+			w.Header().Set("Access-Control-Allow-Methods", "GET, POST, OPTIONS")
+			w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
```

Comment:
I needs these lines on GET and POST as well please

Reply:
I have moved them outside of the switch statement so the code is not duplicated 3x